src/controllers/howDoYouWantToPay.controller.ts

```diff
@@ -0,0 +1,131 @@
+}
+
+@controller(HOW_DO_YOU_WANT_TO_PAY_URI)
+export class HowDoYouWantToPayController extends BaseController {
+
+  public constructor(
+    @inject(SessionService) private session: SessionService,
+    @inject(FormValidator) private validator: FormValidator,
+    @inject(DissolutionService) private service: DissolutionService,
+    @inject(PaymentService) private paymentService: PaymentService,
+    @inject(TYPES.PAY_BY_ACCOUNT_FEATURE_ENABLED) private PAY_BY_ACCOUNT_FEATURE_ENABLED: number
+  ) {
+    super()
+  }
+
+  @httpGet('')
+  public async get(): Promise<string | RedirectResult> {
+    if (!this.PAY_BY_ACCOUNT_FEATURE_ENABLED) {
+      return Promise.reject(new NotFoundError('Feature toggle not enabled'))
+    }
+
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    if (await this.isAlreadyPaid(dissolutionSession, token)) {
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
+
+    return this.renderView(dissolutionSession.howDoYouWantToPayForm)
+  }
+
+  @httpPost('')
+  public async post(@requestBody() body: HowDoYouWantToPayFormModel): Promise<string | RedirectResult>  {
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    if (await this.isAlreadyPaid(dissolutionSession, token)) {
+      return this.redirect(SEARCH_COMPANY_URI)
+    }
+
+    const errors: Optional<ValidationErrors> = this.validator.validate(body, howDoYouWantToPaySchema)
+    if (errors) {
+      return this.renderView(body, errors)
+    }
+
+    const paymentType: PaymentType = body.paymentType!
+
+    this.updateSession(dissolutionSession, paymentType, body)
+
+    return this.redirect(await this.getRedirectURI(paymentType, dissolutionSession))
+  }
+
+  private async renderView(
+    data?: Optional<HowDoYouWantToPayFormModel>,
+    errors?: Optional<ValidationErrors>): Promise<string> {
+    const viewModel: ViewModel = {
+      data,
+      errors
+    }
+
+    return super.render('how-do-you-want-to-pay', viewModel, errors ? StatusCodes.BAD_REQUEST : StatusCodes.OK)
+  }
+
+  private async getRedirectURI(paymentType: PaymentType, dissolutionSession: DissolutionSession): Promise<string> {
+    if (paymentType === PaymentType.ACCOUNT) {
+      return PAY_BY_ACCOUNT_URI
+    }
+
+    return await this.getCreditCardPaymentUrl(dissolutionSession)
+  }
+
+  private async getCreditCardPaymentUrl(dissolutionSession: DissolutionSession): Promise<string> {
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+
+    dissolutionSession.paymentStateUUID = uuidv4()
+
+    const paymentURL: string = await this.paymentService.generatePaymentURL(token, dissolutionSession, dissolutionSession.paymentStateUUID)
+
+    return paymentURL
```

Comment:
you can just `return this.paymentService.generatePaymentURL(token, dissolutionSession, dissolutionSession.paymentStateUUID)`